libs/community/langchain_community/tools/azure_ai_services/translate_tool.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+import logging
+import os
+from typing import Any, Dict, Optional
+
+from langchain_core.callbacks import CallbackManagerForToolRun
+from langchain_core.tools import BaseTool
+
+logger = logging.getLogger(__name__)
+
+
+class AzureTranslateTool(BaseTool):
+    """
+    A tool that interacts with the Azure Translator API using the SDK.
+
+    This tool queries the Azure Translator API to translate text between
+    languages. It requires an API key and endpoint, which can be set up as
+    described in the Azure Translator API documentation:
+    https://learn.microsoft.com/en-us/azure/ai-services/translator/
+    translator-text-apis?tabs=python
+    """
+
+    translate_client: Any = None
+    default_language: str = "en"
+
+    name: str = "azure_translator_tool"
+    description: str = (
+        "A wrapper around Azure Translator API. Useful for translating text between "
+        "languages. Input must be text (str). Ensure to install the "
+        "azure-ai-translation-text package."
+    )
+
+    @classmethod
+    def validate_environment(cls, values: Dict) -> Any:
```

Comment:
Missing model_validator (from pydantic import model_validator)

```suggestion
    @model_validator(mode="before")
    @classmethod
    def validate_environment(cls, values: Dict) -> Any:
```